veps/sig-network/decouple-nad.md

```diff
@@ -0,0 +1,231 @@
+The manifests will be copied from the original repo and converted to go code.
+Enabling the FG (the one mentioned in option 1) will enable the following:
+
+- Skip NAD query in VMI Controller, and ClusterRole permissions in virt-operator
+- Deploy/reconcile NRI in virt-operator  
+
+##### Pros
+
+- No backward compatibility breach
+- Synchronized upgrade, where removed code is replaced with added code in a controlled manner 
+- Allows for opinionated configuration, e.g. optimize to process only virt-launcher pods. 
+
+##### Cons
+
+- Virt-operator currently only deploys internal components and does not reconcile 3rd party tools. Letting it deal with external components may architecturally be a slippery slope that may furthermore contribute to blurring responsibilities between KubeVirt, [CNAO](https://github.com/kubevirt/cluster-network-addons-operator) and [HCO](https://github.com/kubevirt/hyperconverged-cluster-operator).
+It will be hard to reason why virt-operator should deploy network-resources-injector, but not other network dependencies such as multus, sriov-operator/device-plugin, various CNIs etc.
+- Adds a significant maintenance burden to KubeVirt project.
+- NAD related RBAC rules remain in virt-operator though bound to network-resources-injector.
+- The objective of decoupling is not achieved.
+
+> [!NOTE] The rest of this document assumes Option #1
+
+## API Examples
+
+### NetworkAttachmentDefinitions
+
+**SR-IOV Network:**
+
+```yaml
+apiVersion: k8s.cni.cncf.io/v1
+kind: NetworkAttachmentDefinition
+metadata:
+  name: sriov-net-attach-def
+  annotations:
+    k8s.v1.cni.cncf.io/resourceName: intel.com/sriov_net
+spec:
+  config: |
+    {
+      "cniVersion": "0.3.1",
+      "type": "sriov",
+      "vlan": 100
+    }
+```
+
+
+### Expected virt-launcher Pod (After network-resources-injector)
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: virt-launcher-testvm-xxxxx
+  annotations:
+    # KubeVirt applied annotation
+    k8s.v1.cni.cncf.io/networks: '[{"name":"sriov-net-attach-def","namespace":"default"}]'
+spec:
+  containers:
+  - name: compute
+    resources:
+      requests:
+        # network-resources-injector applied request
+        intel.com/sriov_net: "1"
+      limits:
+        # network-resources-injector applied limit
+        intel.com/sriov_net: "1"
+```
+
+## Scalability
+
+The network-resources-injector approach provides several scalability advantages over the current implementation:
+
+- Caching of NADs avoids redundant API calls.
+- By default runs in 2 replicas, but can be scaled to run in multiple replicas, thus highly available.
+
+## Update/Rollback Compatibility
+
+Existing VMs in clusters opting in with the feature gate will have to deploy network-resources-injector before upgrade, so that new VMs and migrated VMs will be able to consume custom network resources.
```

Comment:
```suggestion
Existing VMs in clusters opting in with the feature gate will must deploy network-resources-injector before upgrade, so that new VMs and migrated VMs will be able to consume custom network resources.
```